ompi/mca/pml/ob1/pml_ob1.c

```diff
@@ -394,12 +395,6 @@ int mca_pml_ob1_add_procs(ompi_proc_t** procs, size_t nprocs)
         return rc;
     }
 
-    OBJ_CONSTRUCT(&reachable, opal_bitmap_t);
```

Comment:
good catch. I would however, move the initialization of the `reachable` array after the `bml_add_procs` check). Something like this patch
```diff
diff --git a/ompi/mca/pml/ob1/pml_ob1.c b/ompi/mca/pml/ob1/pml_ob1.c
index c8d0df3e24..031ae9f625 100644
--- a/ompi/mca/pml/ob1/pml_ob1.c
+++ b/ompi/mca/pml/ob1/pml_ob1.c
@@ -382,11 +382,6 @@ int mca_pml_ob1_add_procs(ompi_proc_t** procs, size_t nprocs)
     if(nprocs == 0)
         return OMPI_SUCCESS;
 
-    OBJ_CONSTRUCT(&reachable, opal_bitmap_t);
-    rc = opal_bitmap_init(&reachable, (int)nprocs);
-    if(OMPI_SUCCESS != rc)
-        return rc;
-
     /* make sure remote procs are using the same PML as us */
     if (OMPI_SUCCESS != (rc = mca_pml_base_pml_check_selected("ob1",
                                                               procs,
@@ -394,16 +389,16 @@ int mca_pml_ob1_add_procs(ompi_proc_t** procs, size_t nprocs)
         return rc;
     }
 
+    if (NULL == mca_bml.bml_add_procs) {
+        return OMPI_ERR_UNREACH;
+    }
+
     OBJ_CONSTRUCT(&reachable, opal_bitmap_t);
     rc = opal_bitmap_init(&reachable, (int)nprocs);
     if (OMPI_SUCCESS != rc) {
         return rc;
     }
 
-    if (NULL == mca_bml.bml_add_procs) {
-        return OMPI_ERR_UNREACH;
-    }
-
     rc = mca_bml.bml_add_procs (nprocs, procs, &reachable);
     OBJ_DESTRUCT(&reachable);
     if (OMPI_SUCCESS != rc) {
```